cmd/node/list.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+)
+
+var listOpts struct {
+	output string
+}
+
+func newListCommand(vp *viper.Viper) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List Hubble nodes",
+		Long:  `List Hubble nodes`,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			hubbleConn, err := conn.New(ctx, vp.GetString("server"), vp.GetDuration("timeout"))
+			if err != nil {
+				return err
+			}
+			defer hubbleConn.Close()
+			return runList(ctx, cmd, hubbleConn)
+		},
+	}
+
+	cmd.Flags().StringVarP(
+		&listOpts.output, "output", "o", "table",
+		`Specify the output format, one of:
+ json:     JSON encoding
+ table:    Tab-aligned columns`)
+	cmd.RegisterFlagCompletionFunc("output", func(_ *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
+		return []string{
+			"json",
+			"table",
+			"wide",
+		}, cobra.ShellCompDirectiveDefault
+	})
+	return cmd
+}
+
+func runList(ctx context.Context, cmd *cobra.Command, conn *grpc.ClientConn) error {
+	req := &observerpb.GetNodesRequest{}
+	res, err := observerpb.NewObserverClient(conn).GetNodes(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	nodes := res.GetNodes()
+	sort.Slice(nodes, func(i, j int) bool {
+		return nodes[i].Name < nodes[j].Name
+	})
+	switch listOpts.output {
+	case "json":
+		bs, err := json.MarshalIndent(nodes, "", "  ")
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprintln(cmd.OutOrStdout(), string(bs))
+		return err
+	case "table", "wide":
+		tw := tabwriter.NewWriter(cmd.OutOrStdout(), 2, 0, 3, ' ', 0)
+		fmt.Fprint(tw, "NAME\tSTATUS\tAGE\tFLOWS/S\tCURRENT/MAX-FLOWS")
+		if listOpts.output == "wide" {
+			fmt.Fprint(tw, "\tVERSION\tADDRESS\tTLS")
+		}
+		fmt.Fprintln(tw)
+
+		for _, n := range nodes {
+			age := "N/A"
+			flowsPerSec := "N/A"
+			if uptime := time.Duration(n.GetUptimeNs()).Round(time.Second); uptime > 0 {
+				age = uptime.String()
+				flowsPerSec = fmt.Sprintf("%.2f", float64(n.GetSeenFlows())/uptime.Seconds())
+			}
+			flowsRatio := "N/A"
+			if maxFlows := n.GetMaxFlows(); maxFlows > 0 {
+				flowsRatio = fmt.Sprintf("%d/%d (%6.2f%%)", n.GetNumFlows(), maxFlows, (float64(n.GetNumFlows())/float64(maxFlows))*100)
+			}
+			version := "N/A"
+			if v := n.GetVersion(); v != "" {
+				version = v
+			}
+			fmt.Fprintf(
+				tw, "%s\t%s\t%s\t%s\t%s",
+				n.GetName(), strings.Title(nodeStateToString(n.GetState())), age, flowsPerSec, flowsRatio,
+			)
+			if listOpts.output == "wide" {
+				tls := "N/A"
+				if t := n.GetTls(); t != nil {
+					tls = "Disabled"
+					if t.Enabled {
+						tls = "Enabled"
+					}
+				}
+				fmt.Fprintf(tw, "\t%s\t%s\t%s", version, n.GetAddress(), tls)
+			}
+			fmt.Fprintln(tw)
+		}
+		return tw.Flush()
+	default:
+		return fmt.Errorf("unknown output format: %s", listOpts.output)
+	}
+}
+
+func nodeStateToString(state relaypb.NodeState) string {
```

Comment:
i think we may want to upcase these values

Reply:
This is done with `strings.Title` at call site. Or do you mean we should title case the values here?

Reply:
Yeah I was suggesting that we display (or always represent) the enum values in all-caps. Not super important